File: 3075.cs
public class Solution {
    public long MaximumHappinessSum(int[] happiness, int k) {
        Array.Sort(happiness);
        Array.Reverse(happiness);
        long tot = 0;
        for (int i = 0; i < k; i++) {
            if (happiness[i] > i) {
                tot += (happiness[i] - i);
            }
            else {
                break;
            }
        }
        return tot;        
    }
}

File: 3075.py
class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        happiness.sort()
        happiness.reverse()
        tot = 0
        for i in range(k):
            tot += max(happiness[i] - i, 0)
            print(tot)
        return tot
